ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -111,9 +113,10 @@ public JoinNode(
         Optional.empty()
     );
 
-    this.schema = buildJoinSchema(joinKey, left, right);
     this.joinType = requireNonNull(joinType, "joinType");
     this.joinKey = requireNonNull(joinKey, "joinKey");
+    this.schema = buildJoinSchema(joinKey, left, right);
```

Comment:
nit: Why did you move this line?

Reply:
I moved it after the `joinKey` was check for null, otherwise we'll get an NPE inside the `buildJoinSchema`.